From e74df62e09595d32ef8c52a359fcfb29f432d7e2 Mon Sep 17 00:00:00 2001
From: Alex Stewart <alex.stewart@ni.com>
Date: Tue, 23 Jan 2024 15:57:22 -0500
Subject: [PATCH] CMakeLists: find gpr library

The ni_grpc_device_server must be linked to the grpc-provided gpr.so
library during compilation.

Modify the CMakeLists logic to find and link gpr.so.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6c966a2b..67074c65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,6 +18,7 @@ endif()
 if(CMAKE_CROSSCOMPILING)
   find_program(_PROTOBUF_PROTOC protoc)
   find_program(_GRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
+  find_library(_GRPC_GPR gpr)
 
   if(NOT _GRPC_DEVICE_NILRT_LEGACY_TOOLCHAIN)
     find_package(gRPC REQUIRED)
@@ -441,6 +442,7 @@ endif()
 target_link_libraries(ni_grpc_device_server
    ${_REFLECTION}
    ${_GRPC_GRPCPP}
+   ${_GRPC_GPR}
    ${_PROTOBUF_LIBPROTOBUF}
    ${CMAKE_DL_LIBS}
    nlohmann_json::nlohmann_json
